### PR TITLE
PATCH add processing to pd in cw

### PR DESCRIPTION
**Pull request by Orta21**
Ref. metriport/metriport-internal#1350

Ticket: #1350

### Dependencies

- Upstream: none
- Downstream: none

### Description

Add prcoesing when doing PD in CW

### Testing

_[Regular PRs:]_

- Local
  - [x] Get it to set pd processing
- Production
  - [ ] monitor

Check each PR.

### Release Plan

- :warning: Points to `master`
- [ ] Merge this

### CODE REVIEW

File: packages/api/src/command/medical/patient/create-patient.ts [hunk 1]
```diff
@@ -18,7 +15,6 @@ export type PatientCreateCmd = PatientNoExternalData & Identifier;
 
 export const createPatient = async (
   patient: PatientCreateCmd,
-  forceCommonwell?: boolean,
```
Comment (Orta21) [inline packages/api/src/command/medical/patient/create-patient.ts:21]: Was never being used

File: packages/api/src/external/commonwell/patient-shared.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class PatientDataCommonwell extends PatientExternalDataEntry {
   constructor(
     public patientId: string,
     public personId?: string | undefined,
-    public status?: LinkStatus | undefined,
+    public discoverStatus?: LinkStatus | undefined,
```
Comment (leite08) [inline packages/api/src/external/commonwell/patient-shared.ts:36]: CW doesn't have the concept of Patient Discovery, we've been calling it LinkStatus (thus the type and other functions like `getLinkStatusCW`, even some internal endpoints).

I suggest we leave as is and revisit this when we're revamping the patient state/status. Unless you want to make the case for this update and we can discuss (happy to take this to Slack to speed it up).

**Review by leite08 — APPROVED** [review]
Approved considering we leave the patient's CW link status as is. Happy to discuss otherwise.

File: packages/api/src/external/commonwell/patient-external-data.ts [hunk 3]
```diff
@@ -97,3 +97,59 @@ export const setCommonwellId = async ({
...
+
+/**
+ * Sets the CommonWell (CW) integration status on the patient.
+ *
+ * @param patientId The patient ID @ Metriport.
+ * @param cxId The customer ID @ Metriport.
+ * @param status The status of integrating/synchronizing the patient @ CommonWell.
+ * @returns
+ */
+export const setPatientDiscoveryStatus = async ({
```
[the remark this answers is not in the document]
Comment (leite08) [reply packages/api/src/external/commonwell/patient-external-data.ts:109]: We talked about this [here](https://metriport.slack.com/archives/C05UXMPAVS8/p1712019091204999?thread_ts=1711980937.032029&cid=C05UXMPAVS8) and decided to leave as is so we can fix this with https://github.com/metriport/metriport-internal/issues/1653

**Review by leite08 — APPROVED** [review]

**Pull request merged**